### PR TITLE
Creates initial version of Sentry Connector

### PR DESCRIPTION
**Pull request by luizalfredo23**

### CODE REVIEW

**Comment by luizalfredo23** [comment]
[[doitsa] Notificar erro no envio/consulta de remessa usando o Sentry](https://trello.com/c/WuVFOtrB/1081-doitsa-notificar-erro-no-envio-consulta-de-remessa-usando-o-sentry)

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# Supported Mule Runtime Versions: 
+${project.devkitVersion}
+# New Features and Functionality
+(default value: Initial version) 
+Example: Added support for Query. http://github.com/mulesoft/sentry/issues/61
+# Closed Issues in this release 
+(include issue link when possible)
+Example: Fixed problem with lost telefone on Bulk Update (CLDCONNECT-123).
+# Known Issues in this release
+Example: Session ID expires after  30 days. http://github.com/mulesoft/sentry/issues/62
```
Comment (hprange) [inline CHANGELOG.md]: @luizalfredo23 this is the default content generated for this file, right? We need to enter the proper release information for this connector. Or leave it blank while no release has been published.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+For released connectors you can download them from the update site in Anypoint Studio. 
+Open Anypoint Studio, go to Help → Install New Software and select Anypoint Connectors Update Site where you’ll find all avaliable connectors.
+
+#Usage
+For information about usage our documentation at http://github.com/mulesoft/sentry.
+
+# Reporting Issues
+
+We use GitHub:Issues for tracking issues with this connector. You can report new issues at this link http://github.com/mulesoft/sentry/issues.
```
Comment (hprange) [inline README.md]: @luizalfredo23 this is the default content generated for this file, right? We need to enter the proper README information for this connector. You have to provide adequate documentation if you want to make this project open source.

File: src/main/java/br/com/doit/connector/sentry/ConnectionManagementStrategy.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	public void connect(@ConnectionKey String dsn) throws ConnectionException {
+		try {
+			client = SentryClientFactory.sentryClient(dsn, factory);
+		} catch (InvalidDsnException exception) {
+			throw exception;
+		}
+	}
+
+	/**
+	 * Desconnect client
```
Comment (hprange) [inline src/main/java/br/com/doit/connector/sentry/ConnectionManagementStrategy.java]: @luizalfredo23 small typo.

File: src/main/java/br/com/doit/connector/sentry/SentryConnector.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+import org.mule.api.annotations.Config;
+import org.mule.api.annotations.Connector;
+import org.mule.api.annotations.Processor;
+import org.mule.api.annotations.param.Optional;
+
+import io.sentry.SentryClient;
+
+@Connector(name="sentry", friendlyName="Sentry Connector")
+public class SentryConnector {
```
Comment (hprange) [inline src/main/java/br/com/doit/connector/sentry/SentryConnector.java]: @luizalfredo23 we need to write proper documentation if we want to make this project open source. What you think about writing Javadocs for this class? We can take a look at other plugins for inspiration.

File: src/main/java/br/com/doit/connector/sentry/SentryConnector.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	@Processor(friendlyName = "Send exception")
+	public void sendException(Exception exception, @Optional HashMap<String, Object> extraInfoMap) {
+		
+		SentryClient client = strategy.client;
+		
+		if(extraInfoMap != null){
+			for (Entry<String, Object> entry : extraInfoMap.entrySet()) {
+				client.addExtra(entry.getKey(), entry.getValue());
+			}
+		}
```
Comment (hprange) [inline src/main/java/br/com/doit/connector/sentry/SentryConnector.java]: @luizalfredo23 we need to set a code format standard before we publish the code.

**Review by hprange — CHANGES_REQUESTED** [review]
@luizalfredo23 implementation is good. We need to improve the docs according to the comments. Let's take a look at this together.

**Review by hprange — APPROVED** [review]

**Pull request merged**